code/frameworks/react-webpack5/package.json

```diff
@@ -53,13 +53,13 @@
     "@types/node": "^16.0.0"
   },
   "devDependencies": {
-    "jest-specific-snapshot": "^8.0.0",
-    "typescript": "~4.9.3"
+    "jest-specific-snapshot": "^8.0.0"
   },
   "peerDependencies": {
     "@babel/core": "^7.22.0",
     "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
-    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0"
+    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0",
+    "typescript": "*"
```

Comment:
what? why?!

This will force all storybook users (using react) to install typescript as a dependency.
Users, that aren't using typescript will be really annoyed!

Reply:
no, it won't happen because it is set to optional= true,

actually, your project needs typescript, because of docgen, uses ts checker to get meta about component.  so it is not for dev, to solve this we need to use peerDependencies and. set optional true, if final does not provide typescript, your  package will provide it

Reply:
FWIW `peerDependenciesMeta` is supported by npm, Yarn 1, Yarn 2+ and PNPM, which surprised me.

Reply:
Yes  peerDependenciesMeta field was introduced in npm version 7, so inherently all other package managers will support

Reply:
I see, then I drop my objection. Thanks for clarifying @chakAs3

Reply:
@ndelangen you can merge it, please? anyway I have it in my 2 PRs but ideally should be in separate PR for everybody to understand the reason behind the change